### PR TITLE
Fix Workbench shutdown RuntimeError and GlobalFigureManager tests

### LINKED ISSUE #24050: Workbench shutdown shows `RuntimeError ...QTableWidget has been deleted`
### Expected behavior
Workbench shutdowns quietly.
### Actual behavior
`RuntimeError: wrapped C/C++ object of type QTableWidget has been deleted` is shown.

### Steps to reproduce the behavior
Open Workbench.
Close Workbench with `X`.
See console for error.

### Platforms affected
Windows

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**
Fixes the RuntimeError _sometimes_ shown on Windows on shutdown.

<details>
<summary>Some details for the error cause</summary>
The reason was the unhandled `DictionaryAction.Clear` was being sent from the Figure's ObservableDictionary to the Figure Manager. This would default to updating all figures. This update would be triggered on Workbench shutdown. 
The error showing was a race condition between whether Qt will free the widget first, or the plots will be updated first.

`DictionaryAction.Clear` is now properly handled and triggers `destroy` on all live figures for proper disposal.

</details>
<br>

A small refactoring was done in `globalfiguremanager.py`.

Tests have been added for the `GlobalFigureManager` and `GlobalFigureManagerObserver` -> `workbench.test_globalfiguremanager.test_globalfiguremanager`

**To test:**
Run `ctest` with `-R workbench` and check that all Workbench tests pass.
Open Workbench and plot a few things, try out all the options

Fixes #24050

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/plotting/globalfiguremanager.py [hunk 1]
```diff
@@ -97,19 +117,13 @@ def destroy(cls, num):
...
+        current_fig_manager.canvas.mpl_disconnect(current_fig_manager._cidgcf)
 
-        # There must be a good reason for the following careful
-        # rebuilding of the activeQue; what is it?
-        oldQue = cls._activeQue[:]
-        cls._activeQue = []
-        for f in oldQue:
-            if f != manager:
-                cls._activeQue.append(f)
+        cls.remove_manager_if_present(current_fig_manager)
```
Comment (DTasev) [inline qt/applications/workbench/workbench/plotting/globalfiguremanager.py]: I'd like some feedback on this - do you think this is a good change? The comment leads to believe that something more is being done - but it seems the list is simply being deep copied without the current active manager.

File: qt/applications/workbench/workbench/plotting/globalfiguremanager.py [hunk 2]
```diff
@@ -97,19 +117,13 @@ def destroy(cls, num):
...
+        current_fig_manager.canvas.mpl_disconnect(current_fig_manager._cidgcf)
 
-        # There must be a good reason for the following careful
-        # rebuilding of the activeQue; what is it?
-        oldQue = cls._activeQue[:]
-        cls._activeQue = []
-        for f in oldQue:
-            if f != manager:
-                cls._activeQue.append(f)
+        cls._remove_manager_if_present(current_fig_manager)
```
Comment (DTasev) [inline qt/applications/workbench/workbench/plotting/globalfiguremanager.py]: I'd like some feedback on this - do you think this is a good change? The comment leads to believe that something more is being done - but it seems the list is simply being deep copied without the current active manager.

**Comment by DTasev** [comment]
OSX and Ubuntu build failures are unrelated - the fix for the Ubuntu test will be merged once it's fixed on master

File: qt/applications/workbench/workbench/plotting/globalfiguremanager.py (hunk 2 above)
Comment (mantid-roman) [reply qt/applications/workbench/workbench/plotting/globalfiguremanager.py]: This makes sense to me.

**Review by mantid-roman — APPROVED** [review]
👍

**Pull request merged**